### PR TITLE
Batch mode for nevergrad for reproducability of results plus no more futures as keys

### PR DESCRIPTION
**Pull request by elaude-pf**
# Improved Augmented Lagrangian Optimization Implementation

### TL;DR

Refactored the augmented Lagrangian optimization implementation to use a more efficient batch processing approach and added proper seeding for reproducibility.

### What changed?

- Replaced dictionary-based future tracking with a list-based approach for better management of concurrent evaluations
- Added proper seeding with `seed_util.seed_everything(123)` for reproducible results
- Implemented batch mode option in the Nevergrad settings
- Fixed constraint scaling by removing the 100x multiplier for edge rotational transform
- Improved future handling by separating completion detection from result processing
- Added random state preservation for the parametrization
- Increased the number of workers from 4 to 17 in the toy example
- Adjusted bounds reduction factor from 0.98 to 0.95 in MHD stable stellarator example

### How to test?

1. Run the augmented Lagrangian toy example to verify the new implementation works correctly
2. Test the optimization examples with both batch mode enabled and disabled
3. Verify that seeded runs produce consistent results

### Why make this change?

The previous implementation had inefficient handling of concurrent evaluations and lacked proper seeding for reproducibility. The new approach provides better management of futures, eliminates the need for try-except blocks, and enables batch processing mode for potentially faster optimization in certain scenarios. The constraint scaling adjustment ensures more accurate constraint representation.

### CODE REVIEW

**Comment by elaude-pf** [comment]
* **#47** <a href="https://app.graphite.dev/github/pr/proximafusion/constellaration/47?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/proximafusion/constellaration/47?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/proximafusion?ref=merge-queue-instructions-in-stack-comment&prId=47).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: optimization_examples/augmented_lagrangian_toy_example.py [hunk 1]
```diff
@@ -27,10 +26,11 @@ def objective_constraints(x: jnp.ndarray) -> tuple[jnp.ndarray, jnp.ndarray]:
 
 def run(x0: jnp.ndarray):
     objective, constraints = objective_constraints(x0)
-    num_workers = 4
+    num_workers = 17
     budget = 100
     max_time = None
 
+    seed_util.seed_everything(123)
```
Comment (alexbauer-pf) [inline optimization_examples/augmented_lagrangian_toy_example.py:33]: Can't find this in `optimization/augmented_lag...` should it be there, too, with the seed as an input param?

File: src/constellaration/optimization/augmented_lagrangian_runner.py [hunk 2]
```diff
@@ -93,11 +93,8 @@ def objective_constraints(
                 constraints = jnp.array(
                     [
                         metrics.aspect_ratio - aspect_ratio_upper_bound,
-                        100
```
Comment (alexbauer-pf) [inline src/constellaration/optimization/augmented_lagrangian_runner.py:96]: Won't leaving out this factor 100 change the results a lot potentially?

**Review by alexbauer-pf — APPROVED** [review]
Nice one!

File: optimization_examples/augmented_lagrangian_toy_example.py [hunk 3]
```diff
@@ -27,10 +26,11 @@ def objective_constraints(x: jnp.ndarray) -> tuple[jnp.ndarray, jnp.ndarray]:
 
 def run(x0: jnp.ndarray):
     objective, constraints = objective_constraints(x0)
-    num_workers = 4
+    num_workers = 17
```
Comment (amerlo-pf) [inline optimization_examples/augmented_lagrangian_toy_example.py:29]: maybe put a comment to change this or make it a constant on top such that it is clear that you need to set it up

File: optimization_examples/augmented_lagrangian_toy_example.py [hunk 4]
```diff
@@ -43,14 +43,15 @@ def run(x0: jnp.ndarray):
     )
 
     mp_context = multiprocessing.get_context("forkserver")
-    warnings.simplefilter("ignore", cma.evolution_strategy.InjectionWarning)
+    # warnings.simplefilter("ignore", cma.evolution_strategy.InjectionWarning)
```
Comment (amerlo-pf) [inline optimization_examples/augmented_lagrangian_toy_example.py:46]: please remove commented code

File: optimization_examples/augmented_lagrangian_toy_example.py [hunk 5]
```diff
@@ -62,7 +63,7 @@ def run(x0: jnp.ndarray):
         with futures.ProcessPoolExecutor(
             max_workers=num_workers, mp_context=mp_context
         ) as executor:
-            running_evaluations = {}  # maps futures to candidates
+            running_evaluations = []  # list of (future, candidate)
```
Comment (amerlo-pf) [inline optimization_examples/augmented_lagrangian_toy_example.py:66]: please remove commented code

**Review by amerlo-pf — APPROVED** [review]

File: optimization_examples/augmented_lagrangian_toy_example.py (hunk 1 above)
Comment (elaude-pf) [reply optimization_examples/augmented_lagrangian_toy_example.py:33]: It is seeded in the launch-scripts

File: optimization_examples/augmented_lagrangian_toy_example.py (hunk 5 above)
Comment (elaude-pf) [reply optimization_examples/augmented_lagrangian_toy_example.py:66]: That is just a comment (its not un-commented code)

File: src/constellaration/optimization/augmented_lagrangian_runner.py (hunk 2 above)
Comment (elaude-pf) [reply src/constellaration/optimization/augmented_lagrangian_runner.py:96]: Yes. But it was like that in the original experiment. But anyways we have to rerun all experiments with the new code for a revised version
